### DISCUSSION TITLE
linux arm64 support

### DISCUSSION THREAD

**Discussion by asa** [cyberbotics/webots#6079] [Q&A]
**Describe the Bug**
I tried to build on ubuntu 22 lts on arm64 architecture.  There was an issue building an interface element which required moc. The moc bundled with webots in bin/qt/moc appears to be an  x86 executable. I removed this dir and installed moc from qt6, but was unable to complete building the repo. Please advise what we might need to get in place in order to support a different architecture.

**Steps to Reproduce**
1, download webots on an arm64 linux host. 
2) attempt to build using make -jN

**Expected behavior**
The project should build and launch on this architecture.

**Screenshots**
truncated output is like:
```# compiling wren/WbWrenRangeNoise.cpp
# compiling wren/WbWrenRangeQuantization.cpp
# compiling wren/WbWrenShaders.cpp
# compiling wren/WbWrenSmaa.cpp
# compiling nodes/utils/WbWrenVertexArrayFrameListener.cpp
# compiling vrml/WbWriter.cpp
make[1]: *** No rule to make target 'build/release/WbAboutBox.moc.cpp', needed by 'WbAboutBox.moc.o'.  Stop.
make[1]: *** Waiting for unfinished jobs....
# compiling core/WbPosixMemoryMappedFile.cpp
make: *** [Makefile:112: webots_target] Error 2
```

**System**
linux ubuntu 22, aarm64
`Linux hostname 5.15.0-69-generic #76-Ubuntu SMP Fri Mar 17 17:25:19 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux`

**Additional context**

**Comment by omichel** [comment]
Compiling Webots on Linux/arm64 is not officially supported. It is not supposed to work out-of-box. We never tried it. However, it should be possible with some adjustments, e.g., if you take care of using only arm64 libraries. Webots compiles on macOS/arm64, so I guess there is no major roadblock for getting it compiling on Linux/arm64.

**Comment by robertofiguz** [comment]
@asa Were you able to compile it for ARM64 on ubuntu22?

**Comment by up200707458** [comment]
Greetings,

Managed to compile webots with non-optional dependencies for linux arm64 (Asahi Linux Fedora 41).
Maybe it can help others.

A quick summary:
The change from both only supporting ubuntu and x86_64 meant:
- Had to find equivalent packages to the ones where apt was used.
  Those using ubuntu would just not use this component since it's just a listing of packages for fedora.
  I suppose when #6780 goes through (which i have not cross-referenced with the packages I listed) this step is removed.
  
- The build process downloads some binary libraries that when on linux only the x86_64 is available (in fact the Makefiles assume linux ⇒ x86_64).
  Cross referencing the wiki instruction and the x86_64 packages managed to build them in arm64.
  - QT, is still downloaded but the minimum version found to support arm64 (6.7.0)
  - For the rest i documented their build instructions but also included the compressed archives of the compiled libraries the build expects (they are small ~7MiB) for those that don't want to be bothered with the step.

The changes were done so as to be additive on the platform support of webots and not specialize it to arm64 while destroying other build paths. This is the hope that it eases the way for the developers to include official support (at least for ubuntu) of arm64 linux.

The fork can be found in:
- https://github.com/up200707458/webots/

Relevant cross references:
- #6760
- #6069
- #4416
